jupyterhub/app.py

```diff
@@ -2411,37 +2412,50 @@ def ask():
 
         self.log.info("JupyterHub is now running at %s", self.proxy.public_url)
         # register cleanup on both TERM and INT
-        atexit.register(self.atexit)
         self.init_signal()
 
     def init_signal(self):
-        signal.signal(signal.SIGTERM, self.sigterm)
+        loop = asyncio.get_event_loop()  # TODO: should use running loop
```

Comment:
No need for this TODO. `get_event_loop` already gets the current running loop since the asyncio loop is a thread-global singleton

Reply:
Ah, yes, I forgot that one. Originally I wanted to use `asyncio.get_running_loop()`, since `get_event_loop()` wil create one if there is none,  but that method is only available in python >=3.7. So, it's better to stay with `get_running_loop()`.